src/Console/Commands/EventStoreWorker.php

```diff
@@ -76,31 +76,49 @@ private function connectToStream($streams, $callback): void
 
     private function processPersistentStream($eventStore, string $stream): void
     {
+        $onStart = LaravelEventStore::$onStart;
+        $onSuccess = LaravelEventStore::$onSuccess;
+        $onError = LaravelEventStore::$onError;
+        $onFinish = LaravelEventStore::$onFinish;
+
         $eventStore
             ->persistentSubscription($stream, config('eventstore.group'), $this->option('parallel') ?? 1)
-            ->subscribe(function (AcknowledgeableEventRecord $event) {
+            ->subscribe(function (AcknowledgeableEventRecord $event) use ($onStart, $onSuccess, $onError, $onFinish) {
+                $onStart($event);
                 try {
                     $this->dispatch($event);
                     $event->ack();
+                    $onSuccess($event);
                 } catch (\Exception $e) {
                     $this->dumpEvent($event);
                     $event->nack();
+                    $onError($e, $event);
                     report($e);
```

Comment:
Shouldn't both `report` and `dumpEvent` be per `onError`?

Reply:
Done